hoodie-common/src/main/java/com/uber/hoodie/common/model/HoodieTimer.java

```diff
@@ -0,0 +1,60 @@
+/*
```

Comment:
@vinothchandar To calculate time taken for different sections of code in Hoodie, say IOHandle or Compaction etc, we need a generic way for other modules to plugin this later. I see 2 ways : 

1. Add a start() and end() sort of code in every method where such timing is needed. A centralized class can manage start and end tuples and return back the correct time taken. This seems very dirty from a code perspective and it's difficult to maintain as code evolves.
2. Use AOP : https://en.wikipedia.org/wiki/Aspect-oriented_programming. I've used this before for logging but can be used for Timing as well. Then a method can just be annotated to give timing. But this seems like more complexity to bring in the code. 
Spark itself has some metrics that it provides but that's at the Task level which is already available in the SparkUI and can also be used by a client by registering listeners and doesn't give specific insight into hoodie operations. 
Would be good to hear your thoughts before I proceed further..

Reply:
I vote for keeping it simple for now. #1. I don't see a big win between introducing a class and the hand crafted start/end calls per se. Using Spark listeners is an interesting route we could explore, as long as we get some metadata on what each task/stage are..